### PR TITLE
[web] remove common.ChildFocus

### PR DESCRIPTION
**Pull request by gzzhanghao**
This PR eliminates `common.ChildFocus` mixin by extracting `returnFocus` method to `ProxyAppMain` component.

### CODE REVIEW

**Comment by mhils** [comment]
Maybe I'm missing someone, but I don't see that `ProxyAppMain` implements `returnFocus`?

File: web/src/js/components/proxyapp.js [hunk 1]
```diff
@@ -55,6 +55,8 @@ var ProxyAppMain = React.createClass({
         };
     },
     focus: function () {
+        document.activeElement.blur();
+        window.getSelection().removeAllRanges();
```
Comment (mhils) [inline web/src/js/components/proxyapp.js]: What's the point of these two added lines here?

Comment (gzzhanghao) [inline web/src/js/components/proxyapp.js]: @mhils They comes from `ChildFocus.returnFocus`. Not sure if  `activeElement.blur()` is necessary. But for convenience I just moved everything here.

See https://github.com/mitmproxy/mitmproxy/blob/master/web/src/js/components/common.js#L59

**Comment by gzzhanghao** [comment]
@mhils It passes `focus` method as `returnFocus` in context. `ChildFocus.returnFocus` can safely replace the `ProxyAppMain.focus` method. So I think it's not necessary to add a new method which implements the same thing right there. :confused:

**Comment by mhils** [comment]
:+1:

**Pull request merged**